### ISSUE TITLE
Windows support planned?

### ISSUE THREAD

**Issue by fire** [pwnwriter/hysp#9]
I was curious. Is Windows support planned?

**Comment by Azathothas** [comment]
@fire 
Hi, I am the current package maintainer for hysp: [metis-os/hysp-pkgs](https://github.com/metis-os/hysp-pkgs)

To answer you question:  No.
The reason is, the default packages hysp uses are all compiled statically for x86_64 and arm64.
This is done to ensure maximum compatibility and that installed packages will run anywhere without requiring any dependencies.
Compiling windows binaries is neither practical nor feasible.

But, you do have the options.
Theoretically, you could compile hysp for windows and then host your own packages for windows as shown in the documentation.

**Comment by fire** [comment]
What that take? Compile hysp for windows?

**Comment by Azathothas** [comment]
@fire 

yes compile for windows, but @pwnwriter should be able to answer this in more details

**Comment by Azathothas** [comment]
Unfortunately not,
It seems OS is hard-coded:
![image](https://github.com/pwnwriter/hysp/assets/58171889/1961f57c-b08d-4c1b-be1b-cbf9bea6fe8f)
```bash
error[E0433]: failed to resolve: could not find `unix` in `os`
  --> src\commands\install.rs:16:26
   |
16 | use std::{io::Write, os::unix::fs::PermissionsExt};
   |                          ^^^^ could not find `unix` in `os`

error[E0599]: no function or associated item named `from_mode` found for struct `Permissions` in the current scope
   --> src\commands\install.rs:172:36
    |
172 |     let permissions = Permissions::from_mode(0o755); // Set file permissions to be executable for the current user
    |                                    ^^^^^^^^^ function or associated item not found in `Permissions`

Some errors have detailed explanations: E0433, E0599.
For more information about an error, try `rustc --explain E0433`.
error: could not compile `hysp` (bin "hysp") due to 2 previous errors
```

**Comment by pwnwriter** [comment]
Hello,
@fire @Azathothas ,

I've implemented permissions handling only for *nix. I've plans to make it available for windows as well. Please give me some time.

**Comment by pwnwriter** [comment]
Hey there, @fire,

Upon careful consideration regarding the implementation of Hysp for Windows, several important points have emerged:

- Hysp primarily deals with binary packages. However, Windows Firewall can inadvertently remove installed packages, presenting a significant challenge. Personally, I am not accustomed to using Windows, and I'm constrained for time to manage issues related to the Windows Firewall.

- Given that penetration testers generally prefer Linux/Unix systems, the potential benefits of adapting Hysp for Windows appear limited.

Expanding Hysp to Windows presents notable challenges in package management and may not align well with package building procedures. Therefore, I've decided not to pursue the development of Hysp for Windows. Should anyone wish to create a version for Windows, they are welcome to fork the project and make the necessary modifications. I'll gladly direct Windows users to the respective fork or repository.

Thank you for your support. 

Best regards,
PwnWriter
